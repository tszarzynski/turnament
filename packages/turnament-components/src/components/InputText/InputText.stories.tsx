import InputText from "./InputText";

export default {
	component: InputText,
	title: "InputText",
};

export const Default = () => {
	return <InputText value="Text" />;
};

export const Disabled = () => {
	return <InputText disabled={true} value="Text" />;
};
