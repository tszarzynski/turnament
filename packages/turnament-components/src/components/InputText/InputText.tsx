type Props = React.InputHTMLAttributes<HTMLInputElement>;

const InputText = ({
	value,
	onChange,
	className,
	disabled,
	...rest
}: Props) => {
	const variantStyles = "border-black";
	const disabledStyles =
		"border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent";

	return (
		<input
			{...rest}
			disabled={disabled}
			className={`focus:none border-1 border-black font-bold text-lg placeholder:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-1 ${disabled ? disabledStyles : variantStyles} ${className}`}
			type="text"
			value={value}
			onChange={onChange}
			autoComplete="off"
			autoCorrect="off"
		/>
	);
};

export default InputText;
