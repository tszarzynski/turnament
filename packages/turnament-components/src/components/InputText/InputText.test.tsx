import { render } from "vitest-browser-react";
import InputText from "./InputText";

describe("InputText", () => {
	it("renders without error", () => {
		const { asFragment } = render(<InputText />);
		expect(asFragment()).toMatchSnapshot();
	});
});
