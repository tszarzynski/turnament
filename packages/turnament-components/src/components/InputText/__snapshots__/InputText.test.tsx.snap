// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputText > renders without error 1`] = `
<DocumentFragment>
  <input
    autocomplete="off"
    autocorrect="off"
    class="focus:none border-1 border-black font-bold text-lg placeholder:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-1 border-black undefined"
    type="text"
  />
</DocumentFragment>
`;
